src/kibana/directives/aria_expanded.js

```diff
@@ -0,0 +1,23 @@
+define(function (require) {
+  require('modules')
+    .get('kibana')
+    .directive('ariaExpanded', function () {
+      return {
+        restrict: 'A',
+        link: function ($scope, $elem, attrs) {
+          $elem.bind('click', function () {
+            if ($elem[0].getAttribute('aria-expanded') === 'true') {
```

Comment:
Yes the original problem was changing things in the DOM from a controller. With Angular there is a strict separation of business logic, and view logic. [Controllers](https://docs.angularjs.org/guide/controller) are strictly for business logic, and [Directive](https://docs.angularjs.org/guide/directive) are made to manipulate the DOM.

That is why I suggested you remove the original code.


Regarding the dynamic `[aria-label]`'s

[Here](https://github.com/elasticsearch/kibana/pull/3129/files#diff-74f05d7354fe601d536f92ab9da38cd8R1) you have a `[aria-label]` with a dynamic value.

I'm saying you should do the same thing with the `[aria-expanded]` and the the directive here is redundant.